akka-actor/src/main/scala/akka/event/Logging.scala

```diff
@@ -1116,7 +1241,18 @@ trait DiagnosticLoggingAdapter extends LoggingAdapter {
    * These values can be used in PatternLayout when [[akka.event.slf4j.Slf4jLogger]] is configured.
    * Visit <a href="http://logback.qos.ch/manual/mdc.html">Logback Docs: MDC</a> for more information.
    */
-  def setMDC(jMdc: java.util.Map[String, Any]): Unit = mdc(if (jMdc != null) jMdc.asScala.toMap else emptyMDC)
+  def setMDC(jMdc: java.util.Map[String, Any]): Unit = {
+    val mdcValue = if (jMdc != null) {
+      val iterator = jMdc.entrySet().iterator()
+      val builder = Map.newBuilder[String, Any]
+      while (iterator.hasNext) {
+        val entry = iterator.next()
+        builder += (entry.getKey -> entry.getValue)
+      }
+      builder.result()
```

Comment:
This is exactly what `jMdc.asScala.toMap` ends up doing, so I prefer the shorter version for readability.